ecosystem/typescript/sdk_v2/tests/unit/hex.test.ts

```diff
@@ -0,0 +1,71 @@
+import { Hex } from "../../src/types/hex";
+
+const mockHex = {
+  withoutPrefix: "007711b4d0",
+  withPrefix: "0x007711b4d0",
+  bytes: new Uint8Array([0, 119, 17, 180, 208]),
+};
+
+test("creates a new Hex instance from bytes", () => {
+  const hex = new Hex({ data: mockHex.bytes });
+  expect(hex.toUint8Array()).toEqual(mockHex.bytes);
+});
+
+test("creates a new Hex instance from string", () => {
+  const hex = new Hex({ data: mockHex.bytes });
+  expect(hex.toString()).toEqual(mockHex.withPrefix);
+});
+
+test("converts hex bytes input into hex data", () => {
+  const hex = new Hex({ data: mockHex.bytes });
+  expect(hex instanceof Hex).toBeTruthy();
+  expect(hex.toUint8Array()).toEqual(mockHex.bytes);
+});
+
+test("converts hex string input into hex data", () => {
+  const hex = Hex.fromString({ str: mockHex.withPrefix });
+  expect(hex instanceof Hex).toBeTruthy();
+  expect(hex.toUint8Array()).toEqual(mockHex.bytes);
+});
+
+test("accepts hex string input without prefix", () => {
+  const hex = Hex.fromString({ str: mockHex.withoutPrefix });
+  expect(hex instanceof Hex).toBeTruthy();
+  expect(hex.toUint8Array()).toEqual(mockHex.bytes);
+});
+
+test("accepts hex string with prefix", () => {
+  const hex = Hex.fromString({ str: mockHex.withPrefix });
+  expect(hex instanceof Hex).toBeTruthy();
+  expect(hex.toUint8Array()).toEqual(mockHex.bytes);
+});
+
+test("converts hex string to bytes", () => {
+  const hex = Hex.hexInputToBytes({ hexInput: mockHex.withPrefix });
+  expect(hex instanceof Uint8Array).toBeTruthy();
+  expect(hex).toEqual(mockHex.bytes);
+});
+
+test("converts hex bytes to string", () => {
+  const hex = Hex.hexInputToString({ hexInput: mockHex.bytes });
+  expect(typeof hex).toEqual("string");
+  expect(hex).toEqual(mockHex.withPrefix);
+});
+
+test("converts hex bytes to string without 0x prefix", () => {
+  const hex = Hex.hexInputToStringWithoutPrefix({ hexInput: mockHex.withPrefix });
+  expect(hex).toEqual(mockHex.withoutPrefix);
+});
+
+test("throws when parsing invalid hex char", () => {
+  expect(() => Hex.fromString({ str: "0xxyz" })).toThrow();
+});
+
+test("throws when parsing hex of length zero", () => {
+  expect(() => Hex.fromString({ str: "0x" })).toThrow();
```

Comment:
can add the message it throws. i.e here it throws `Hex string is too short, must be at least 1 char long, excluding the optional leading 0x.`